### PR TITLE
🎓 Lists Topic loops on lists

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
Looping on lists

### How
Mostly examples

### Additional Notes
1. Not sure if this is the best place to introduce `range`
2. May want to move the subsequent sections to the topic *after* references
    * This topic is already getting long
    * It makes sense to talk about aliases after references
3. Need to go and check for indentation consistency throughout the lecture content since colab likes 2 spaces. This is a setting on colab that can be changed, but I only did it now and then realized the issue.

### CODE REVIEW

File: site/topics/lists/lists.rst [hunk 1]
```diff
@@ -112,13 +112,173 @@ Mutability
...
+* Using the ``range`` function would help in this situation as we can use it to loop over each index in ``haystack``
+* The only catch is that we need to actually index ``haystack``
+
+.. code-block:: python
+    :linenos:
+
+    def index_of(needle, haystack):
+        haystack_length = len(haystack)
+        for i in range(haystack_length):
+            if haystack[i] == needle:
```
Comment (twentylemon) [inline site/topics/lists/lists.rst:229]: [enumerate](https://docs.python.org/3/library/functions.html#enumerate) is specifically for this case.

File: site/topics/lists/lists.rst [hunk 2]
```diff
@@ -112,13 +112,173 @@ Mutability
...
+    #   3 d
+    #   4 e
+
+
+* Here there is no need to actually index the list since the ``element`` variable already has the value ``another_list[i]``
+
+.. note::
+
+    Some people love using ``enumerate``, others feel it is unnecessary since they do not see indexing the list as an
+    imposition. Neither is more correct than the other --- use what you are comfortable with.
```
Comment (twentylemon) [inline site/topics/lists/lists.rst:274]: Well. The argument is intent. The goal is to loop through the list, adding a layer of indirection through `range` is needlessly complicating. _Simplicity_ is the argument -- I'd argue that is indeed _more correct_.

Comment (jameshughes89) [reply site/topics/lists/lists.rst:274]: perhaps _neither is wrong_.

Comment (jameshughes89) [reply site/topics/lists/lists.rst:274]: The argument _against_ using it here, in intro CS, is avoiding another keyword that is particularly python-ie.

Comment (twentylemon) [reply site/topics/lists/lists.rst:274]: It's a function, not a keyword. But maybe fair enough. Probably too early to introduce tuples and multiple returns.

I'd still maybe change the wording here. `enumerate` is more appropriate, but delves into some python features you have not (and will not?) touched on in the course.

Comment (jameshughes89) [reply site/topics/lists/lists.rst:274]: Oh, how about, yeet the "note" in question?

Comment (twentylemon) [reply site/topics/lists/lists.rst:274]: Eh. Also fine.

File: site/topics/lists/lists.rst [hunk 3]
```diff
@@ -268,11 +268,6 @@ Looping Over Indices and Enumerate
 
 * Here there is no need to actually index the list since the ``element`` variable already has the value ``another_list[i]``
 
-.. note::
-
-    Some people love using ``enumerate``, others feel it is unnecessary since they do not see indexing the list as an
-    imposition. Neither is more correct than the other --- use what you are comfortable with.
-
```
Comment (jameshughes89) [inline site/topics/lists/lists.rst]: Note cut

**Pull request merged**